### ISSUE TITLE
Make cholesky_inverse documentation mention that batching is supported

### ISSUE THREAD

**Issue by Fitz13009** [pytorch/pytorch#72334]
### 🚀 The feature, motivation and pitch

Dear PyTorch developers,

hi, I am a ML researcher and a PyTorch programmer working heavily in the prediction of Gaussian distribution with full coverariance matrix. In my work, I need to use NN to predict the cholesky decomposition of the covariance matrix, and then operate it quite often. 

One of my operation is to compute the inverse of the covariance matrix in a batched manner. I found an implementation [using the cholesky](https://pytorch.org/docs/stable/generated/torch.cholesky_inverse.html), which makes me excited. But later I found it only supports 2d input, so I would like to ask a new feature to make it support batched input [..., 2d]. 

Thanks a lot in advance.

Best,
Ge Li

cc @brianjo @mruberry @jianyuh @nikitaved @pearu @walterddr @IvanYashchuk @xwang233 @Lezcano

**Comment by Lezcano** [comment]
Even though we should mentoin this somewhere, you can imitate cholesky inv and cholesky solve via  `linalg.solve_triangular`. This is because `LL^TX = B` can be solved via two `solve_triangular`s, one with `L` and the other one with `L^T`. Set `B=Id` and you have the inverse. 

As always, I'd very much recommend you not to use the inverse and then multiply, but doing solves agianst hte matrix that you want to multiply to later on.

**Comment by IvanYashchuk** [comment]
Both `torch.cholesky_inverse` and `torch.cholesky_solve` support batched inputs and they use the same backend function so their performance should roughly be the same. It's an oversight in the documentation that it specifies only 2D inputs.
```py
In [1]: import torch

In [2]: a = torch.randn(3, 2, 2, dtype=torch.float64)

In [3]: a = a @ a.mT # make symmetric

In [4]: L = torch.linalg.cholesky(a) # Factorize the batched matrix

In [5]: a_inv = torch.cholesky_solve(torch.eye(2, dtype=L.dtype), L) # compute the batched inverse one way

In [6]: torch.allclose(a_inv, a.inverse())
Out[6]: True

In [7]: a_inv2 = torch.cholesky_inverse(L) # compute the batched inverse the other way

In [8]: torch.allclose(a_inv2, a_inv)
Out[8]: True
```

**Comment by Fitz13009** [comment]
Hi, thanks a lot for your answer and support. 
I did not know these cholesky's functions by heart before. 
Maybe one other question: Do the cholesky_inverse and cholesky_solve functions offer same numerical stabilities? 
We know that normally doing the solve() instead of inverse() can offer better numerical stabilities.

**Comment by IvanYashchuk** [comment]
As Mario mentioned above if the intention is to multiply the inverse by some other matrix, often it should be more efficient to use the solve function to do that. Internally, `torch.cholesky_inverse(L)` computes the result by calling `torch.cholesky_solve(eye(L.shape[0]), L)`.

Leaving performance aside, I don't know whether the accuracy would differ much. This [answer on Math.StackExchange](https://math.stackexchange.com/a/2398452) suggests that the forward error of using the inverse explicitly or solving the linear system is about the same.

**Comment by Fitz13009** [comment]
Thanks again for your answer, I have no more doubt. Please feel free to close this issue, or maybe after the document has been updated.  :+1: 
Best,
Ge Li
